build-tools/packages/build-tools/src/typeValidator/compatibility.ts

```diff
@@ -26,26 +26,49 @@ type requireAssignableTo<_A extends B, B> = true;
  * This will strip some type branding information which ideally would be kept for stricter checking, but without it, const enums show up as breaking when unchanged.
  */
 export const typeOnly = `
-// See 'build-tools/src/type-test-generator/compatibility.ts' for more information.
+// See 'build-tools/packages/build-tools/src/typeValidator/compatibility.ts' for more information.
+type ValueOf<T> = T[keyof T];
+type WellKnownSymbols = OnlySymbols<ValueOf<typeof Symbol>>;
+type OnlySymbols<T> = T extends symbol ? T : never;
+// Omit (replace with never) a key if it is a symbol thats not a well known symbol from global Symbol.
+type FilterKey<P> = symbol extends P
```

Comment:
`symbol extends P`? What do symbols extend from? Am I understanding right that this means something like "if the key's type is so 'fundamental' that even symbols extend from it, then the key should be considered for type testing"? (The rest of the ternary(ies) I do understand).

Reply:
So the only case where `symbol extends P` is true is when `P` is `symbol` itself (and not any custom unique symbol), correct? In that case we want to allow the key, so use `P`. Else, use `P` if it's not a symbol at all or it is a well-known symbol, and use `never` if it's a custom unique symbol to disallow them.

Reply:
I have updated the code and docs a bit to make this more clear.